### ISSUE TITLE
Problem with esm module

### ISSUE THREAD

**Issue by O4epegb** [TypeStrong/ts-loader#874]
So, there is a library [nanobus](https://github.com/choojs/nanobus), which is using another library [remove-array-items](https://github.com/mreinstein/remove-array-items).

`Nanobus` in its dist file imports `remove-array-items`  like that:

```
var splice = require('remove-array-items')
```

`Nanobus` has 2 dist files, commonjs and es module. It exports itself like that in module file:

```
function removeItems() ...
// actual code there
[...]

export default removeItems;
```

In my project, when I compile everything with webpack and ts-loader, I get a runtime error that `splice` is not a function. And indeed it is not a function, but and object which `default` field is needed function.

My `.tsconfig` looks like this:

```
{
  "compilerOptions": {
    "baseUrl": "./src",
    "module": "es2015",
    "moduleResolution": "node",
    "target": "es5",
    "lib": ["dom", "es2015"],
    "jsx": "react",
    "esModuleInterop": true,
    "allowSyntheticDefaultImports": true,
    "noEmit": true,
    "sourceMap": true,
    "noImplicitAny": true,
    "allowJs": true,
    "experimentalDecorators": true,
    "importHelpers": true
  },
  "include": ["./src/**/*"]
}
```

Webpack loader config:

```
{
          test: [/\.tsx?$/, /\.js?$/],
          use: {
            loader: 'ts-loader',
            options: {
              transpileOnly: true,
              compilerOptions: {
                noEmit: false
              }
            }
          },
          exclude: /node_modules/
        },
```

I tried many different options, tried to remove or add some flags, include and exclude files, but nothing changes for some reason. 

Generated webpack export of `remove-array-items` looks like that

```
/* harmony default export */ __webpack_exports__["default"] = (removeItems);
```

Import inside `nanobus` like that

```
var splice = __webpack_require__(/*! remove-array-items */ "./node_modules/remove-array-items/dist/remove-array-items.esm.js")
```

So, am I doing something wrong or there is no way to compile that stuff?
The only thing I cannot change is `module: "es2015"` I think, because I need dynamic imports.

**Comment by BlooJeans** [comment]
To add to this:
nanobus depends on `"remove-array-items": "^1.0.0"`, which previously resolved to `1.0.0`, and as of 27 days ago, now resolves to `1.1.0`, which is when they changed how the function was exported, causing the `splice` function to not be the actual thing exported, but instead an object.

To fix the issue in your app, you might be able to force it to use version 1.0.0 by adding 
`"remove-array-items": "1.0.0"` to your apps package.json

**Comment by O4epegb** [comment]
Yes, thank you! I know that though, and with `yarn.lock` it is quite easy to force this dependency. I just want to know how to properly deal with, if that possible.

**Comment by BlooJeans** [comment]
Also opened an issue: https://github.com/choojs/nanobus/issues/37

I don't know anything about ts-loader, so i can't help you there, sorry!  Found you via googling the splice issue.   Thanks for opening this issue

**Comment by mreinstein** [comment]
Instead of just routing around the problem, it would help everyone if we fix the upstream issue. pull requests welcome on https://github.com/mreinstein/remove-array-item

**Comment by mreinstein** [comment]
fixed in remove-array-items npm version 1.1.1

**Comment by O4epegb** [comment]
@mreinstein But it is not a problem and we are not routing around the problem. You did everything right (except for semver), and it should work, but for some reason it don't

**Comment by mreinstein** [comment]
fair enough. This is probably something having to do with the bundle/build process. I'm going to publish remove-array-items@2.0 which will adopt the esm support. (basically the same behavior encountered in 1.1.0)

**Comment by O4epegb** [comment]
Yep, thank you!)
Now I just want to understand why compilation is not working!

**Comment by johnnyreilly** [comment]
Me neither 😉

But I'm afraid I only have so much time I can spare to look into issues. Sorry ♥️🌻

**Comment by mreinstein** [comment]
I would suggest not closing this issue if there is a problem. By closing the ticket it essentially hides the problem.

**Comment by johnnyreilly** [comment]
Agreed. But I'm not certain there is a problem with ts-loader. If it can be clearly demonstrated there is then great. But it's not clear to me that's the case
